common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -0,0 +1,125 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.embeddedwebview;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+import android.view.MotionEvent;
+import android.view.View;
+import android.webkit.WebView;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStrategy;
+
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Serve as a class to do the OAuth2 auth code grant flow with Android embedded web view.
+ */
+public class EmbeddedWebViewAuthorizationStrategy extends AuthorizationStrategy {
+    private static final String TAG = StringExtensions.class.getSimpleName();
+    private WebView mWebView;
+    private String mStartUrl;
+
+    /**
+     * Perform the authorization request in the embedded web view and return the authorization result.
+     *
+     * @param request authorization request
+     * @return AuthorizationResult
+     */
+    public AuthorizationResult requestAuthorization(@NonNull final AuthorizationRequest request) {
+        loadStartUrl();
+        throw new UnsupportedOperationException("Not implemented yet.");
+    }
+
+    /**
+     * Constructor of EmbeddedWebViewAuthorizationStrategy.
+     *
+     * @param activity Authentication activity
+     * @param request  Authorization request
+     * @throws UnsupportedEncodingException
+     * @throws ClientException
+     */
+    public EmbeddedWebViewAuthorizationStrategy(@NonNull final Activity activity,
+                                                @NonNull final AuthorizationRequest request)
+            throws UnsupportedEncodingException, ClientException {
+        //TODO validate auth request in OAuth2Strategy.
+        createWebView(activity, request);
+        mStartUrl = request.getAuthorizationStartUrl();
+    }
+
+    /**
+     * Set up the web view configurations.
+     * @param activity  AuthenticationActivity
+     * @param request AuthorizationRequest
+     */
+    private void createWebView(final Activity activity, final AuthorizationRequest request) {
+        // Create the Web View to show the page
+        mWebView = (WebView) activity.findViewById(activity.getResources().getIdentifier("webView1", "id",
+                activity.getPackageName()));
+        mWebView.getSettings().setUserAgentString(
+                mWebView.getSettings().getUserAgentString() + AuthenticationConstants.Broker.CLIENT_TLS_NOT_SUPPORTED);

```

Comment:
I know this is ported from ADAL... this constant name is unfortunate... since we're actually telling the server that we support PKAuth.  Would be great to fix this now.